### ISSUE TITLE
Properties for R448A

### ISSUE THREAD

**Issue by jurgenolivier** [CoolProp/CoolProp#1849]
Am I making an error or are properties for R448A (Honeywell HFO blend) not yet available in CoolProp? I can, however, obtain properties for R449A (DuPont HFO blend).

Using <PropsSI("D","T",273.15,"Q",0,"R448A.mix")> I get the following error:

Initialize failed for backend: "?", fluid: "R448A.mix" fractions "[ 1.0000000000 ]"; error: Could not match the binary pair [29118-24-9,354-33-6] - for now this is an error. :   PropsSI("D","T",273.15,"Q",0,"R448A.mix")

However, with "R449A.mix" I get a result of 1197.703 [...kg/m³].

I was expecting R448A properties to be calculated given their proximity to R449A, so am not sure if this is because they're not yet available, my OS, the way I'm trying to access the properties or some error on my side?

**CoolProp Version: 6.3.0
**Windows 10 Pro
**Trying to access via Excel for now

[R448A.xlsx](https://github.com/CoolProp/CoolProp/files/3497697/R448A.xlsx)
[R449A.xlsx](https://github.com/CoolProp/CoolProp/files/3497698/R449A.xlsx)

**Comment by jurgenolivier** [comment]
In terms of resolving above issues with calculating fluid properties for R448A in CoolProp, am I correct in saying that a lack of binary pair interaction parameters for R1234ze[E] is the main challenge?

In this regard there's an interesting paper by Bell, I & Lemmon, E 2016, 'Automatic Fitting of Binary Interaction Parameters for Multi-ﬂuid Helmholtz-Energy-Explicit Mixture Models', Journal of Chemical & Engineering Data, 61, pp. 3752-3760

Is there a way to contact the authors of this paper (I realise Ian is active member on this forum) to provide the raw data to calculate the parameters, or is it possible to obtain a sample calculation?

Am in conversation with Honeywell and might be possible to obtain the raw data, but will be necessary to know quite accurately what is required?

**Comment by jmcarter17** [comment]
I also have this problem as some refrigration systems in some supermarkets now use R448A (and R449A and R450A).
For R449A, our calculation for the saturated temperature from pressure works, but not for R448A (nor for R450A), due the inability to match the binary pair [29118-24-9,354-33-6] (or [29118-24-9,811-97-2] for R450A).

If I understand, @ianhbell mentionned several time that Interaction parameters are missing for this binary pair, in this case you'll need REFPROP for R448A, or to get someone (not me) to implement the necessary parameters in CoolProp.

So my question is, Is there anyone working on adding the interaction parameters for those pairs?
As this becomes more and more relevant since the refrigerant exists in real world settings, I would like to help make it work. However, I don't have the slightest idea how to provide those interactions parameters.

**Comment by dvd101x** [comment]
Hi, I'm under the same dilemma, by fixing errors of binary pairs one by one I came with the following running code on v6.6.0.

```py
import CoolProp.CoolProp as CP
from scipy.constants import convert_temperature

# If you need to overwrite use the following
# CP.set_config_bool(CP.OVERWRITE_BINARY_INTERACTION, True)
CP.apply_simple_mixing_rule("R1234ze(E)", "R125", "linear")
CP.apply_simple_mixing_rule("R1234ze(E)", "R1234yf", "linear")
CP.apply_simple_mixing_rule("R1234ze(E)", "R134a", "linear")
CP.apply_simple_mixing_rule("R32", "R125", "linear")
CP.apply_simple_mixing_rule("R32", "R1234yf", "linear")
CP.apply_simple_mixing_rule("R32", "R134a", "linear")
CP.apply_simple_mixing_rule("R32", "R1234ze(E)", "linear")
CP.apply_simple_mixing_rule("R1234yf", "R125", "linear")
CP.apply_simple_mixing_rule("R134a", "R125", "linear")
CP.apply_simple_mixing_rule("R134a", "R1234yf", "linear")

fluid = "R448A.mix"
density = CP.PropsSI("D", "T", convert_temperature(-50, "C", "K"), "Q|gas", 1, fluid)
print(f"density of R448A at -50 C an dew point {density:.2f} kg / m^3")

# If you need to find the name of a particular CAS
# print(CP.get_fluid_param_string("754-12-1", 'name'))
```

> **warning:** didn't validate if 'linear' method is acceptable, just focused on getting running code.

**Comment by ianhbell** [comment]
For R-448A, the models for all but R-125/1234ze(E) are available in the literature and just need to be implemented in CoolProp.

**Comment by ibell** [comment]
For R-125/1234ze(E) someone would have to do experimental measurements. Once that is complete, I could quickly put together a model. The parameters for R-125/1234yf *might* be a useful substitute, but experiments are needed to be sure

**Comment by jmcarter17** [comment]
I have been hoping for phase enveloppe and mixing parameters for R448A, R449A and R450A for many years now.
I decided to finally purchase a REFPROP license.

Now, trying to use the REFPROP backend in CoolProp for those mixtures, I now get the following:
`ValueError: Initialize failed for backend: "REFPROP", fluid: "R448A" fractions "[ 1.0000000000 ]"; error: Could not load these fluids: R448A : PropsSI("P","T",253.15,"Q",1,"REFPROP::R448A")`

Now, I have testes other mixtures with the REFPROP backend successfully, so now I am wondering what I am doing wrong.
Sorry if this is not the place for such a question, but I can't find the info I am looking for.

**Comment by henningjp** [comment]
I don't see R448A in the pre-defined mixtures list.  Try a mixture with Composition: 26% R32, 26% R125, 21% R134a,20% R1234yf, 7% R1234ze.

**Comment by henningjp** [comment]
Ah, just read Ian's comment above.  The last component is not implemented.

**Comment by jmcarter17** [comment]
Right, but with the REFPROP backend, the R448A.MIX file is there, but it fails for some reason (Could not load these fluids).

**Comment by ibell** [comment]
@jmcarter17 Please show/explain what you did. Otherwise it is impossible to figure out how to help you.

**Comment by jmcarter17** [comment]
I installed REFPROP (after purchasing a personal license).
I compiled the Dynamic Library.
I tested the library with some calls using PropsSI, which worked.
Example : `PropsSI("P", "T", 253.15, "Q", 1, "REFPROP::R404A")` worked.
But then I tried a call to PropsSI to get Saturated Temperature <-> Pressure conversion working for R448A (R449A and R450A), and I got the message: 

`ValueError: Initialize failed for backend: "REFPROP", fluid: "R448A" fractions "[ 1.0000000000 ]"; error: Could not load these fluids: R448A : PropsSI("P","T",253.15,"Q",1,"REFPROP::R448A")`

System information:
MacOS 15.3.1
Chip : M2 Pro
Installed REFPROP using wine.
Copied all the files in the REFPROP-cmake folder
Built the librefprop.dylib from there.
Moved the library to somewhere in my path.
Added the os.environ['RPPREFIX'] = "/Applications/REFPROP/" 
Made sure all the files are in "/Applications/REFPROP/" 

What I find weird is that R404A works with the REFPROP Backend, but for R448A, it doesn't

**Comment by ibell** [comment]
This stuff is tricky to get right on mac. Honestly, with mac, I think I would recommend calling REFPROP via ctREFPROP, otherwise the path issues can be quite tricky, like you are experiencing. In any case, you need instead to use the ALTERNATIVE_REFPROP_PATH as indicated in the docs:  http://coolprop.org/coolprop/REFPROP.html#path-issues.  I think I might add additionally the SETPATHdll calling in CoolProp, which is yet another alternative.

**Comment by jmcarter17** [comment]
Hi @ibell thanks for the answer.
Good news, it seems to work via ctREFPROP.
The use of ALTERNATIVE_REFPROP_PATH however did not seem to help.
I think that making it work in CoolProp on Mac (with SETPATHdll or something) might help.
I was not able to make it work in CoolProp yet, even if it works with ctREFPROP.
Would be willing to help troubleshoot anyone who would like to see how to make it work on a Mac.

**Comment by ibell** [comment]
I am in the process of adding a new environment variable that will "win" and will be used for ALL REFPROP initialization, and will be used to call the SETPATH function after the library loads. This is what I should have done before and it will allow the use of predefined mixture files as well. The only constraint is that you will be REQUIRED to have the standard folder layout.

**Comment by jmcarter17** [comment]
Sounds reasonable enough.
What is the standard folder layout?
I don't think I have changed it. I simply copied the folder from the REFPROP installation to some path on my Mac.
Thanks again!

**Comment by ibell** [comment]
See https://github.com/CoolProp/CoolProp/pull/2513
